test/tasks/SingleMultisigTask.t.sol

```diff
@@ -246,17 +253,25 @@ contract SingleMultisigTaskTest is Test {
             )
         );
         vm.expectRevert(expectedRevertMessage);
-        localMultisigTask.simulateRun(taskConfigFilePath);
+        localMultisigTask.simulate(taskConfigFilePath);
     }
 
     function testExecuteWithSignatures() public {
         uint256 snapshotId = vm.snapshotState();
         (, Action[] memory actions) = runTask();
         addrRegistry = multisigTask.addrRegistry();
         multisigTask.processTaskActions(actions);
-        bytes memory callData = multisigTask.getMulticall3Calldata(actions);
-        bytes memory dataToSign = multisigTask.getEncodedTransactionData(multisigTask.parentMultisig(), callData);
-        address multisig = multisigTask.parentMultisig();
+
+        address[] memory allSafes = MultisigTaskTestHelper.getAllSafes(multisigTask.root());
+        uint256[] memory allOriginalNonces = MultisigTaskTestHelper.getAllOriginalNonces(allSafes);
+        bytes[] memory allCalldatas = multisigTask.transactionDatas(actions, allSafes, allOriginalNonces);
+
+        address rootSafe = allSafes[allSafes.length - 1];
+        bytes memory rootSafeCalldata = allCalldatas[allCalldatas.length - 1];
+        uint256 rootSafeNonce = allOriginalNonces[allOriginalNonces.length - 1] - 1; // The task has already run so we decrement the nonce by 1.
```

Comment:
I think that you could do with a `_getRootSafeData() internal returns(address rootSafe, uint256 rootNonce, bytes rootCalldata)` function.